### ISSUE TITLE
reorder by color

### ISSUE THREAD

**Issue by lexelby** [inkstitch/inkstitch#378]
Let's say you want to combine multiple designs together or multiple copies of the same design in one canvas (e.g. to mass produce small patches).  You'll have a lot of unnecessary color changes as it sews each design individually.

Let's add an extension that groups colors together to reduce color changes.  For example, let's say you have a design with 3 colors:

* Red
* Yellow
* Orange

If you placed four copies of this design in one canvas it would sew like this:

* Red
* Yellow
* Orange
* Red
* Yellow
* Orange
* Red
* Yellow
* Orange
* Red
* Yellow
* Orange

This extension would reorder it like this:

* Red
* Red
* Red

**Comment by wwderw** [comment]
One thing to be aware of when doing multiples of **different** files.  The re-ordering may not be good.

For instance, one file may sew red 2nd, while another file needs red to be sewn 4th otherwise it may expose travelling stitches or the red may cover up something that it shouldn't.

I could see this for multiples of the same file, but it may need to be more involved to do different files nested in one frame.

**Comment by lexelby** [comment]
Good point!

**Comment by lexelby** [comment]
I'm thinking if that's the case, the user would simply not run this extension.

**Comment by DreadKnight** [comment]
Was thinking of the same thing, as I made small flags a month ago and I would end up making an "industrial" file, where I could have to duplicate one flag 6 times and regroup the colors and since my machine doesn't supports trimming for the same color, I would have to hide the running stitches.
If I had to adjust the original design, it was a bit tedious to duplicate/move the flags since inkscape linked duplicate is not really supported before export, then group the colors and adjust the running stitches etc.
Simply having an extension to move and "group" the extensions by same name/order would be nice.

**Comment by lexelby** [comment]
@DreadKnight Interesting that you mention the linked duplicate feature, which I believe refers to the `<svg:use>` tag in the SVG file.  If Ink/Stitch supported `<svg:use>`, would that make things easier for you?

**Comment by tyrosinase** [comment]
FWIW, I use Inkscape's "select same" and then just group the result. Pretty sure there's no "select same fill OR stroke color" but at least it knocks things down to two select's. Also of course it doesn't work on stuff that's already grouped with other-color stuff, but I try to limit my groupings for that reason.

**Comment by DreadKnight** [comment]
@lexelby Yes, I guess so (I'm not too technical about svg files, but googled that and it would seem so).

@tyrosinase Wasn't aware of that feature in Inkscape, might give it a go, as I found it in the edit menu.
